### PR TITLE
Remove _pthread_isduecanceled in favor of ->cancel. NFC.

### PR DESCRIPTION
**Pull request by kleisauke**
~~Also simplify the wait in slices logic a bit, while we're at it.~~ I'll make another PR for that.

Split from PR #15603.

### CODE REVIEW

File: system/lib/pthread/pthread_testcancel.c [hunk 1]
```diff
@@ -8,15 +8,11 @@
 #include "pthread_impl.h"
 #include <pthread.h>
 
-int _pthread_isduecanceled(struct pthread* pthread_ptr) {
```
Comment (sbc100) [inline system/lib/pthread/pthread_testcancel.c:11]: Sorry to be a pain but can you make the PR *only* about the removal of this function.. I really like simple PRs that do just one thing.

Comment (kleisauke) [reply system/lib/pthread/pthread_testcancel.c:11]: No problem, I thought this simplification was too small for a dedicated PR, but maybe that doesn't matter. Reverted with commit 1bb8198bab97b47d1b36d15d8184dcb73f095ba9.

File: system/lib/libc/musl/src/thread/__timedwait.c [hunk 2]
```diff
@@ -73,7 +69,7 @@ int __timedwait_cp(volatile int *addr, int val,
 	    pthread_self()->cancelasync == PTHREAD_CANCEL_ASYNCHRONOUS) {
 		double sleepUntilTime = emscripten_get_now() + msecsToSleep;
 		do {
-			if (_pthread_isduecanceled(pthread_self())) {
+			if (pthread_self()->cancel) {
```
Comment (sbc100) [inline system/lib/libc/musl/src/thread/__timedwait.c:72]: As a followup we should replace `pthread_self` with `__pthread_self` in all our musl-internal code. 

Also, `pthread_t self = __pthread_self();` would probably make sense here.

**Review by sbc100 — APPROVED** [review]
Nice!

**Pull request merged**